### DISCUSSION TITLE
Thrift protocol error when uploading results

### DISCUSSION THREAD

**Discussion by erikjv** [Ericsson/codechecker#3562] [Q&A]
Hi,

We're using CodeChecker on a C++ codebase, and for certain areas we get a protocol error when storing results on the server:

```
thrift.protocol.TProtocol.TProtocolException: Unexpected character: b'<'
```

My guess is that this is the start of some HTML error response, so is there a way to show the full response? The server doesn't log any errors, and it works fine for other parts of the same codebase. It happens as soon as we enable clang-tidy, even when the config file only has `clang-diagnostic-*` enabled.

Any help would be very welcome!

**Comment by csordasmarton** [comment]
Are you using `nginx`, `apache` or some different technology in top of the `CodeChecker server`? In this case you have to check their logs.

**Comment by quanahwu** [comment]
I've been using "CodeChecker store --verbose debug ..." and "CodeChecker server --verbose debug ...".

**Comment by olologin** [comment]
I had similar issue recently with nginx because nginx by default limits http request size, and when we upload results our request is HUGE.
I found it by patching thrift sources temporarily, that place which throws the exception, and instead of showing just first incorrect symbol I printed 100 first symbols. This is how I discovered error code and nginx message.
I am not sure if such changes should be upstreamed, because they are in thrift, and because they can potentially cause some security problems!?
